src/state_migration/common/state_migration.rs

```diff
@@ -78,18 +80,29 @@ impl<BS: Blockstore + Send + Sync> StateMigration<BS> {
         }
 
         let cache = MigrationCache::new(NonZeroUsize::new(10_000).expect("infallible"));
+        let num_threads = std::env::var("FOREST_STATE_MIGRATION_THREADS")
+            .ok()
+            .and_then(|s| s.parse().ok())
+            // At least 3 are required to not deadlock the migration. Don't use all CPU,
+            // otherwise the migration will starve the rest of the system.
+            .unwrap_or_else(|| 3.max(num_cpus::get() / 2));
+
         let pool = rayon::ThreadPoolBuilder::new()
             .thread_name(|id| format!("state migration thread: {id}"))
-            .num_threads(3) // minimum needed, more doesn't increase performance in any way
+            .num_threads(num_threads)
             .build()?;
 
-        let (state_tx, state_rx) = crossbeam_channel::bounded(1);
-        let (job_tx, job_rx) = crossbeam_channel::bounded(1);
+        let (state_tx, state_rx) = crossbeam_channel::bounded(30);
+        let (job_tx, job_rx) = crossbeam_channel::bounded(30);
```

Comment:
nit: why are the channels bound specifically to `30`?

Reply:
Magic number, same as cache set to 10k. Setting it to higher value, just like with cache sizes, may increase the memory footprint.

Reply:
as long as it works. I would probably make it a constant with a little comment instead, but not critical.